src/init/starship.xsh

```diff
@@ -0,0 +1,15 @@
+import random
+import string
+
+
+def starship_prompt():
+    last_cmd = __xonsh__.history[-1] if __xonsh__.history else None
+    status = last_cmd.rtn if last_cmd else 0
+    jobs = len(__xonsh__.all_jobs)
+    duration = round((last_cmd.ts[1] - last_cmd.ts[0]) * 1000) if last_cmd else 0
+    return $(::STARSHIP:: prompt --status=@(status) --jobs=@(jobs) --cmd-duration=@(duration))
+
+
+$PROMPT = starship_prompt
+$STARSHIP_SHELL = "xonsh"
+$STARSHIP_SESSION_KEY = "".join(random.choice(string.ascii_letters + string.digits) for _ in range(16))
```

Comment:
I'm getting a `NameError: name 'random' is not defined` here.

Reply:
Reproduced on my end.

I'm pretty confused by xonsh's behavior here. I've reduced some of the surprising behavior to a minimal example and asked about it on the xonsh gitter: https://gitter.im/xonsh/xonsh?at=60e7d763951c58084ecb7108

Reply:
As a work around you could use the uuid module from the python std library. I think this would be enough we just need a unique id for each session 
```python
srt(uuid.uuid4())
# or
uuid.uuid4().hex
```

Reply:
Done. Thanks for that suggestion.